sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+the linker defined symbol ``_DYNAMIC``. Some dynamic linker's use this
+address or another reserved symbol to find its own dynamic structure
+without without having yet processed their own relocation entries. For
+AArch64 entries one and two in the ``.got`` are also reserved.
+
+For AArch64 the linker defined ``_GLOBAL_OFFSET_TABLE_`` symbol should
+be the address of the first global offset table entry in the ``.got``
+section.
+
+See `Sample code sequences for code models`_ for code model for
+examples of how to access the ``.got``.
+
+Function Addresses
+^^^^^^^^^^^^^^^^^^
+
+Direct function calls are those where the name of the called function
+is known at compile time. The PC-relative BL instruction may be used
+for all direct function calls, whether absolute or position
+independent.
+
+Indirect function calls are those where the address of the function is
+in a pointer. Appropriate code is used to load the value of the
+pointer into a register, as for other data, and then a BLR instruction
+is used.
+
+In statically linked code the address of a function is always its real
+address. However in dynamically linked environments references to the
+address of a function from an executable file and its shared objects
+might not resolve to the same value.
+
+References from within shared objects to a function address will
+normally be resolved by the dynamic linker to the virtual address of
+the function itself.
+
+References from within a non position independent executable file to a
+function address defined in a shared object will normally be resolved
+by the static linker to the address of the `Procedure Linkage Table`_
+(PLT) for that function within the executable file. If the address is
+in a writeable location the linker may use a dynamic relocation
+instead.
+
+When both shared objects and a non position independent executable use
+the address of a function both the shared library and the application
+must use the same address for the functions for address comparisons to
+work as expected. If the static linker uses a PLT entry for a function
+as the address for a function then it must place the address of the
+PLT for the function in the dynamic symbols table entry for the
+function. This will result in a dynamic symbol table entry with a
+section index of ``SHN_UNDEF``, a type ``STT_FUNC`` and a non-zero
+``st_value``. A reference to this symbol will be resolved by the
+dynamic linker to the address PLT for the function in the executable.
+
+The ``R_<CLS>_JUMP_SLOT`` relocations defined in AAELF64_ are
+associated with PLT entries. These entries are used for direct
+function calls rather than for references to function addresses. These
+relocations do not use the special symbol value described
+above. Otherwise a very tight endless loop would result.
+
+In a position independent executable all function addresses have
+dynamic relocations so no indirection via the PLT entry is necessary.
```

Comment:
I'll see what I can do.

Reply:
> all function addresses have dynamic relocations

When the function address is local, PC-relative instructions can be used to avoid dynamic relocations.

Reply:
Ack.